libcron/include/libcron/SortableQueue.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <vector>
+#include "Task.h"
+
+namespace libcron
+{           
+    template<typename LockType>
+    class SortableQueue
+    {
+        public:
+            const std::vector<Task>& get_tasks() const
+            {
+                return c;
+            }
+            
+            std::vector<Task>& get_tasks()
+            {
+                return c;
+            }
+            
+            size_t size() const noexcept
+            {
+                return c.size();
+            }
+            
+            bool empty() const noexcept
+            {
+                return c.empty();
+            }
+            
+            void push(Task& t)
+            {
+                c.push_back(std::move(t));
+            }
+            
+            void push(Task&& t)
+            {
+                c.push_back(std::move(t));
+            }
+            
+            void push(std::vector<Task>&& tasks_to_insert)
```

Comment:
I happily admit I've not fully mastered move sematics yet, so I'm wondering what is the the reason for an r-value reference here? The vector is only used to copy from, no ownership is transferred, everything is copied. Also, see the next comment which relates to this argument.

Reply:
Hmmm, I agree with you, there indeed no reason, a simple reference will do the job.